### PR TITLE
Potential partial fix for broken socket exceptions

### PR DESCRIPTION
**Pull request by annagrin**
Wrap `serveRequest` calls in `Chain.capture` with an error handler to 

- make sure it runs in an error zone that logs exceptions and continues execution
- capture all async stack traces

Closes: https://github.com/dart-lang/webdev/issues/1418

### CODE REVIEW

File: dwds/lib/src/utilities/shared.dart [hunk 1]
```diff
@@ -72,9 +77,16 @@ Future<HttpServer> startHttpServer(String hostname, {int port}) async {
   return httpServer;
 }
 
-/// Throws an [ExceptionDetails] object if `exceptionDetails` is present on the
+void serveHttpRequests(Stream<HttpRequest> requests, Handler handler,
```
Comment (grouma) [inline dwds/lib/src/utilities/shared.dart:80]: This probably should have a doc comment.

**Review by grouma — APPROVED** [review]
We should expand the description to include why we think this will resolve the socket exception issue.

File: dwds/lib/src/utilities/shared.dart (hunk 1 above)
Comment (annagrin) [reply dwds/lib/src/utilities/shared.dart:80]: will add, thanks!

Comment (annagrin) [reply dwds/lib/src/utilities/shared.dart:84]: done.

**Review by natebosch — APPROVED** [review]

**Pull request merged**